### PR TITLE
Immutable configuration properties not shown by Actuator

### PR DESCRIPTION
**Pull request by polarbear567**
After testing, I found the problem was on the serializer modifier, so I made this change.

### CODE REVIEW

**Comment by polarbear567** [comment]
@snicoll  Could you please spare time to look at it? I don't know why this kind of failure occurs. I tested all cases of jarcommandit locally, and they can pass.

**Comment by snicoll** [comment]
@polarbear567 I am sure you have the best intention but there is no need to at me for a review, the PR is open and someone from the team will have a look to it when time permits.

**Comment by polarbear567** [comment]
Hi, @snicoll I think you misunderstood me. It's not that this PR needs your review, but that there is a problem with the CI build of spring-boot(like this: https://ci.spring.io/builds/93431). I want to ask you to help me to have a look this problem. After that, I tried again and passed.

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java [hunk 1]
```diff
@@ -231,10 +233,7 @@ else if (item instanceof List) {
 		@Override
 		public Object findFilterId(Annotated a) {
 			Object id = super.findFilterId(a);
-			if (id == null) {
-				id = CONFIGURATION_PROPERTIES_FILTER_ID;
-			}
-			return id;
+			return (id != null) ? id : CONFIGURATION_PROPERTIES_FILTER_ID;
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java]: This is unrelated to the task at hand, please refrain from polishing code even if it's in the same class.

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java [hunk 2]
```diff
@@ -312,6 +311,13 @@ public void serializeAsField(Object pojo, JsonGenerator jgen, SerializerProvider
 		}
 
 		private boolean isReadable(BeanDescription beanDesc, BeanPropertyWriter writer) {
+			// if the class has the @ConstructorBinding annotation or
+			// one constructor of this class has the @ConstructorBinding annotation,
+			// we can ignore the setter method.
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java]: Unfortunately, that's not what the code should be doing. If constructor binding is to be used, only the parameters of the matching constructor should be used to determine the properties. Accessors (getters) shouldn't be checked at all.

**Comment by polarbear567** [comment]
@snicoll Thank you for your opinion. I'd like to work this. It's complicated but interesting.

**Comment by polarbear567** [comment]
Sorry to click the re-review button first. At present, it is judged according to the parameters in the constructor. There are many situations for the application of this annotation. I have considered all situations as much as possible, but I am not sure whether there will be any omissions.

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java (hunk 1 above)
Comment (polarbear567) [reply spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java]: Yes, I will change this back, this is a temporary modification I made to re trigger CI builder. If you have any other methods that can be triggered repeatedly, please let me know.

**Comment by snicoll** [comment]
@polarbear567 every commit you push sends a notification to a large number of watchers. In the interest of keeping those to an acceptable level, please refrain from pushing isolated commits and, in particular, commits to trigger a new build. The build should work for you locally before submitting changes and we can figure out on our side what led to the build failure when reviewing.

**Comment by polarbear567** [comment]
@snicoll Sure, I will notice this. And I'm not sure how to rebuild cli build, because my local build was successful

**Comment by snicoll** [comment]
@polarbear567 thanks for the efforts thus far but I think this is more involved than I thought initially and the lack of tests in this PR makes it hard to track what is covered and what isn't. I've started to roll-out a solution of my own and I've a number of tests that are covering the change. Given the complexity and potential change in other part of the codebase I am going to close this one. Thanks anyway.
